File: MiniJira.Server/Entities/Issue.cs
namespace MiniJira.Server.Entities
{
    public class Issue
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Key { get; set; } = null!;
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public string Type { get; set; } = null!;
        public string Priority { get; set; } = null!;
        public string Status { get; set; } = null!;
        public Guid ProjectId { get; set; }
        public Guid ReporterId { get; set; }
        public Guid? AssigneeId { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public User? Reporter { get; set; }
        public User? Assignee { get; set; }
        public Project? Project { get; set; }
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
        public ICollection<IssueAttachment> IssueAttachments { get; set; } = new List<IssueAttachment>();
    }
}

File: minijira.client/src/models/Issue.ts
import { Dayjs } from "dayjs";

export interface Issue {
  id?: string;
  key?: string;
  title?: string;
  description?: string;
  type?: IssueType;
  priority?: IssuePriority;
  status?: IssueStatus;
  projectId?: string;
  projectName?: string;
  reporterId?: string;
  reporterName?: string;
  assigneeId?: string;
  assigneeName?: string;
  createdAt?: Dayjs;
  updatedAt?: Dayjs;
  attachmentIds?: string[];
  attachmentUrls?: string[];
  logs?: string[];
}

export interface ChangeIssueData {
  oldData?: string;
  newData?: string;
}

export interface IssueFilter {
  pageIndex?: number;
  pageSize?: number;
  type?: IssueType;
  priority?: IssuePriority;
  status?: IssueStatus;
  assigneeId?: string;
  projectId?: string;
}

export type IssueType = "bug" | "task" | "story" | "epic";
export type IssueStatus = "todo" | "in_progress" | "in_review" | "done";
export type IssuePriority = "highest" | "high" | "medium" | "low" | "lowest";
